File: apps/web/src/test/fixtures/historian.ts
import type { HistorianStatus } from "@/entities/historian/model/types";

export const historianPersistentFixture: HistorianStatus = {
  enabled: true,
  mode: "persistent",
  status: "connected",
  database: "postgresql/timescaledb",
  writeIntervalMs: 1000,
  telemetrySampleMs: 1000,
  lastSuccessfulWriteAt: "2026-05-21T06:00:00Z",
  fallbackActive: false,
  retentionEnabled: true,
  rawRetention: "30 days",
  downsamplingEnabled: true,
  supportedResolutions: ["raw", "1m"],
  aggregateStatus: "telemetry_history_1m",
  simulationOnly: true,
  safetyDisclaimer: "The historian stores synthetic simulation data only.",
};

export const historianFallbackFixture: HistorianStatus = {
  ...historianPersistentFixture,
  enabled: false,
  mode: "in_memory",
  status: "disabled",
  database: "in-memory",
  fallbackActive: true,
  retentionEnabled: false,
  downsamplingEnabled: false,
  supportedResolutions: ["raw"],
  aggregateStatus: "disabled",
};

File: apps/simulation/internal/model/historian.go
package model

import "time"

type HistorianStatusValue string

const (
	HistorianStatusDisabled              HistorianStatusValue = "disabled"
	HistorianStatusConnected             HistorianStatusValue = "connected"
	HistorianStatusDegraded              HistorianStatusValue = "degraded"
	HistorianStatusUnavailableFallback   HistorianStatusValue = "unavailable_fallback"
	HistorianModeInMemory                string               = "in_memory"
	HistorianModePersistent              string               = "persistent"
	HistorianStoragePostgresTimescale    string               = "postgresql/timescaledb"
	HistorianStorageInMemory             string               = "in_memory"
	HistorianSimulationOnlyDataStatement string               = "The historian stores synthetic simulation data for demo, learning and portfolio purposes only."
)

type HistorianStatus struct {
	Enabled               bool                 `json:"enabled"`
	Mode                  string               `json:"mode"`
	Status                HistorianStatusValue `json:"status"`
	Database              string               `json:"database"`
	WriteIntervalMS       int                  `json:"writeIntervalMs"`
	TelemetrySampleMS     int                  `json:"telemetrySampleMs"`
	LastSuccessfulWriteAt *time.Time           `json:"lastSuccessfulWriteAt,omitempty"`
	LastErrorAt           *time.Time           `json:"lastErrorAt,omitempty"`
	LastErrorMessage      string               `json:"lastErrorMessage,omitempty"`
	FallbackActive        bool                 `json:"fallbackActive"`
	RetentionEnabled      bool                 `json:"retentionEnabled"`
	RawRetention          string               `json:"rawRetention,omitempty"`
	DownsamplingEnabled   bool                 `json:"downsamplingEnabled"`
	SupportedResolutions  []string             `json:"supportedResolutions,omitempty"`
	AggregateStatus       string               `json:"aggregateStatus,omitempty"`
	SimulationOnly        bool                 `json:"simulationOnly"`
	SafetyDisclaimer      string               `json:"safetyDisclaimer"`
}
